flytekit/core/type_engine.py

```diff
@@ -180,6 +184,34 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
         return dc
 
 
+class ProtobufTransformer(TypeTransformer[_proto_reflection.GeneratedProtocolMessageType]):
+    PB_FIELD_KEY = "pb_type"
+
+    def __init__(self):
+        super().__init__("Protobuf-Transformer", _proto_reflection.GeneratedProtocolMessageType)
+
+    @staticmethod
+    def tag(expected_python_type: Type[T]) -> str:
+        return f"{expected_python_type.__module__}.{expected_python_type.__name__}"
+
+    def get_literal_type(self, t: Type[T]) -> LiteralType:
+        return LiteralType(simple=SimpleType.STRUCT, metadata={ProtobufTransformer.PB_FIELD_KEY: self.tag(t)})
```

Comment:
do you want to model this as binary or struct?

Reply:
Haytham said struct was better - something about showing up in the UI.